### ISSUE TITLE
Do not create `{doc-filename}_images` directory

### ISSUE THREAD

**Issue by ronaldtse** [metanorma/metanorma-standoc#53]
Temporary directories should be created in the `${TMPDIR}` rather than in the local directory. Or, we make TMPDIR customizable on build.

**Comment by opoudjis** [comment]
I will create the directory locally if it is HTML output without data-uri encoding, since that is where all images are gathered up. I will create it in /tmp otherwise.

**Comment by opoudjis** [comment]
... Actually, no, I can't do that: the Word doc generation relies on the images being in the local directory. So given that both HTML and DOC rely on local directory images, I think it is enough that I am deleting the temporary directory when finished, per https://github.com/metanorma/metanorma-nist/issues/55

**Comment by ronaldtse** [comment]
Is it possible to generate the word file in TMPDIR and move it to the destination afterwards?

**Comment by opoudjis** [comment]
Too much risk of things breaking, since the Asciidoctor source will be referencing relative links to local files. And I'm demoting this to low priority.

**Comment by ronaldtse** [comment]
@CAMOBAP795 could you help work on this? Thanks!

**Comment by CAMOBAP795** [comment]
In progress

**Comment by opoudjis** [comment]
Yes, correct.

**Comment by CAMOBAP795** [comment]
Currently, I'm testing locally on different `metnorma-*` modules but faced with a problem on https://github.com/metanorma/metanorma-standoc/issues/70

**Comment by CAMOBAP795** [comment]
Blocked by https://github.com/metanorma/metanorma-standoc/issues/70

**Comment by ronaldtse** [comment]
@camobap795 the tests are passing and this can proceed.

**Comment by ronaldtse** [comment]
@w00lf do you have time for this after your current work? Thanks!

**Comment by opoudjis** [comment]
The guilty party is:

isodoc, lib/isodoc/function/to_word_html.rb

````
    # tmp image dir is same directory as @filename
    def tmpimagedir
      @filename + tmpimagedir_suffix
    end

    def rel_tmpimagedir
      Pathname.new(@filename).basename.to_s + tmpimagedir_suffix
    end
````

These pathnames were originally used for temporary copies of images in Metanorma; but I went through code refactoring to get rid of temporary directories and use Tempfile instead. So this is how they are now used.

````
isodoc/lib/isodoc/function/to_word_html.rb:    def tmpimagedir
isodoc/lib/isodoc/function/to_word_html.rb:      @filename + tmpimagedir_suffix
isodoc/lib/isodoc/function/to_word_html.rb:    def rel_tmpimagedir
isodoc/lib/isodoc/function/to_word_html.rb:      Pathname.new(@filename).basename.to_s + tmpimagedir_suffix
isodoc/lib/isodoc/html_function/postprocess.rb:      FileUtils.rm_rf tmpimagedir
isodoc/lib/isodoc/html_function/postprocess.rb:      FileUtils.mkdir tmpimagedir
isodoc/lib/isodoc/html_function/postprocess.rb:      new_full_filename = File.join(tmpimagedir, fname)
isodoc/lib/isodoc/html_function/postprocess.rb:      i["src"] = File.join(rel_tmpimagedir, fname)
isodoc/lib/isodoc/word_function/body.rb:      FileUtils.rm_rf tmpimagedir
isodoc/lib/isodoc/word_function/body.rb:      FileUtils.mkdir tmpimagedir
metanorma-iec/lib/isodoc/iec/word_convert.rb:        FileUtils.rm_rf tmpimagedir
metanorma-iec/lib/isodoc/iec/word_convert.rb:        FileUtils.mkdir tmpimagedir
metanorma-m3d/lib/isodoc/m3d/word_convert.rb:        FileUtils.rm_rf tmpimagedir
metanorma-m3d/lib/isodoc/m3d/word_convert.rb:        FileUtils.mkdir tmpimagedir
````

metanorma-m3d and metanorma-iec create the directory because the normal place they inherit it from doesn't exist in the descendants (the document preface).

The temp directory only survives in one place now in the code: in HTML output, in case we do not use Data URIs, we use it to gather up all the images in the HTML output in one place, as a sister folder to the HTML output. That usage in isodoc/lib/isodoc/html_function/postprocess.rb IS CORRECT, because those images have to go somewhere: they are in fact meant to persist past document creation.

I therefore propose (a) to eliminate the use of tmpimagedir in word output and (b) to close this issue; but I'm happy for you to do testing.

**Comment by ronaldtse** [comment]
> I therefore propose (a) to eliminate the use of tmpimagedir in word output and (b) to close this issue; but I'm happy for you to do testing.

Option (a) would definitely be desirable, but it seems that it no longer applies?

**Comment by opoudjis** [comment]
Right

**Comment by ronaldtse** [comment]
Great, thanks!
